app/src/main/java/com/pileproject/drive/execution/RxObservableProgram.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright (C) 2011-2015 PILE Project, Inc. <dev@pileproject.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.pileproject.drive.execution;
+
+import android.os.Bundle;
+
+import com.pileproject.drive.programming.visual.block.BlockBase;
+
+import java.util.List;
+
+import rx.Observable;
+import rx.Subscriber;
+
+/**
+ * A concrete class of {@link rx.Observable.OnSubscribe}, which generates
+ * a stream of blocks that was executed during a program execution.
+ *
+ * This class is one-off.
+ * Create this instance whenever you want to execute and DO NOT REUSE the instance.
+ * <p/>
+ * The stream emits {@link Bundle} objects for the subscriber.
+ * The bundles contain message type whose key is {@link RxObservableProgram#KEY_MESSAGE_TYPE},
+ * and optionally one message argument whose key is {@link RxObservableProgram#KEY_MESSAGE_ARG}.
+ * The message types are:
+ * <ul>
+ *     <li>{@link RxObservableProgram#MESSAGE_STARTED}: emitted at the beginning of the execution</li>
+ *     <li>{@link RxObservableProgram#MESSAGE_TERMINATED}: emitted when {@link RxObservableProgram#requestRestart()} is called</li>
+ *     <li>{@link RxObservableProgram#MESSAGE_PAUSED}: emitted when {@link RxObservableProgram#requestPause()} is called</li>
```

Comment:
I feel it is helpful to add an explanation for `how we know an execution restarted` with this `PAUSED` message. We can know it by receiving `BLOCK_EXECUTED`.